apisix/plugins/redirect.lua

```diff
@@ -133,7 +135,11 @@ function _M.rewrite(conf, ctx)
         -- TODO： add test case
         -- PR: https://github.com/apache/apisix/pull/1958
         uri = "https://$host$request_uri"
-        ret_code = 301
+        if ngx.req.get_method() == http_post then
```

Comment:
I think it would be better to use whitelist (`GET` OR `HEAD`) instead.

Reply:
sure